src/routes/turnilo_dashboard_routes.py

```diff
@@ -0,0 +1,85 @@
+import re
+from fastapi import APIRouter, Depends, HTTPException, Query
+from typing import List
+from sqlmodel import Session
+from models.turnilo_dashboard import TurniloDashboard
+from services import turnilo_dashboards as td
+
+import constants
+import data.database as db
+
+turnilo_router = APIRouter()
+
+### Dashboards ###
+
+
+def is_valid_query(s: str) -> bool:
+    if not isinstance(s, str):
+        return False
+    if len(s) > 256:
+        return False
+    pattern = r'^[a-zA-Z0-9_-]+$'
+    return bool(re.match(pattern, s))
+
+
+@turnilo_router.get(
+    constants.URL_PATH + "/turnilo/dashboards/",
+    response_model=List[TurniloDashboard],
+    summary="Gets all Turnilo dashboards"
+)
+def turnilo_get_dashboards(db_session: Session = Depends(db.get_session),
+                           shortName: str = Query(None), dataCube: str = Query(None)):
+    if shortName and not is_valid_query(shortName):
+        raise HTTPException(status_code=400, detail=f"Invalid shortName='{shortName}'")
+    if dataCube and not is_valid_query(dataCube):
+        raise HTTPException(status_code=400, detail=f"Invalid dataCube='{dataCube}'")
+    return td.dashboards_get_all(db_session, shortName, dataCube)
```

Comment:
Even if you decide to not to go to the BaseModel approach for query params, it would be great if you protect your DB functions from them. I would aggregate in some kind of Filter function like this: 

```python
f := DashFilter(data_cube, name)
td.dashboards_get_all(db_session, f) # this dashFilter can dump easily filter dicts and if more Queries appear you do not need to refactor all your libraries to accomodate new fields
```

Reply:
N/A, will follow your approach above. Closing.